lib/tpm2_policy.c

```diff
@@ -215,3 +231,112 @@ bool tpm2_policy_get_digest(TSS2_SYS_CONTEXT *sapi_context,
     return true;
 }
 
+bool tpm2_policy_populate_digest_list(char *buf, TPML_DIGEST *policy_list,
+    uint8_t hash_len) {
+
+        unsigned long file_size;
+        bool retval = files_get_file_size_path(buf, &file_size);
+        if (!retval) {
+            return false;
+        }
+        if (file_size != hash_len) {
+            return false;
+        }
+
+        policy_list->digests[policy_list->count].size = hash_len;
+        /* All policy digests are expected to be of same hash len */
+        if (policy_list->count > 0 &&
+            policy_list->digests[policy_list->count].size !=
+            policy_list->digests[policy_list->count - 1].size) {
+            return false;
+        }
+
+        uint16_t policy_digest_size = hash_len;
+        retval = files_load_bytes_from_path(buf, 
+            policy_list->digests[policy_list->count].buffer, &policy_digest_size);
+        if (!retval) {
+            return false;
+        }
+
+        policy_list->count++;
+
+        return true;
+}
+
+bool tpm2_policy_get_digest_list(const char *str_list, TPMI_ALG_HASH hash,
+    TPML_DIGEST *policy_list) {
+
+    uint8_t hash_len = tpm2_alg_util_get_hash_size(hash);
+    const char *str_current;
+    int len_current;
+    int len = strlen(str_list);
+    char buf[1024];
+    bool retval = true;
+
+    do {
+        str_current = str_list;
+        str_list = memchr(str_current, ',', len);
```

Comment:
This code appears to be tokenizing a string on a comma. This should probably use strtok_r. If str_list originates from the argv[] array (think options), it's guaranteed to be NULL terminated.

I think if you strtok_r you can remove the need for the copy into buf below